tests/util.cpp

```diff
@@ -49,6 +49,52 @@ TEST_CASE("String", "[.util]") {
         dxvk::str::tonvss(ss, std::string("Longer-Than-Short-String-Longer-Than-Short-String-Longer-Than-Short-String"));
         REQUIRE_THAT(ss, SizeIs(64));
     }
+
+    SECTION("std::string") {
+        REQUIRE(dxvk::str::fromnullable(nullptr) == std::string());
+        REQUIRE(dxvk::str::fromnullable("") == std::string());
+        REQUIRE(dxvk::str::fromnullable("string") == std::string("string"));
+    }
+
+    SECTION("split") {
+        REQUIRE(dxvk::str::split<std::set<std::string>>("", std::regex(",")).size() == 0);
+
+        auto result = dxvk::str::split<std::set<std::string>>("foo,bar,baz", std::regex(","));
+
+        REQUIRE(result.size() == 3);
+        REQUIRE(result.contains("foo"));
+        REQUIRE(result.contains("bar"));
+        REQUIRE(result.contains("baz"));
+    }
+
+    SECTION("CaseInsensitiveCompare") {
+        dxvk::str::CaseInsensitiveCompare<std::string_view> compare;
+
+        REQUIRE(compare("aa", "aa") == false);
```

Comment:
PS: you can shorten this and the following test case with `GENERATE` (https://github.com/jp7677/dxvk-nvapi/blob/master/tests/nvapi_sysinfo.cpp#L201), but also ok to keep it like this.